Sources/Sentry/SentryTracer.m

```diff
@@ -303,7 +308,8 @@ - (void)deadlineTimerFired
         }
     }
 
-    [self finishWithStatus:kSentrySpanStatusDeadlineExceeded];
+    _finishStatus = kSentrySpanStatusDeadlineExceeded;
+    [self finishInternal];
```

Comment:
The logic in the tracer is so complicated 🤦. I'm only complaining. We should simplify this at some point, but not in this PR.